src/Npgsql/NpgsqlActivitySource.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Npgsql.Internal;
+using System;
+using System.Diagnostics;
+using System.Net;
+
+namespace Npgsql
+{
+    static class NpgsqlActivitySource
+    {
+#if NET5_0_OR_GREATER
```

Comment:
Add `System.Diagnostics.DiagnosticSource` 5.0.1 for downlevel frameworks and enable this for all target frameworks?

Reply:
Not sure there is a lot of value in bringing this to older tfm's. Except .NET framework (and we don't really support it since 5.0) there is only .NET core 3.1 left, which is supported only until the end of 2022. Maybe @roji has a different opinion?

Reply:
I just saw that the csproj already has TFM-conditional PackageReferences: https://github.com/npgsql/npgsql/blob/a7dba56ff9b444f9e48fc6ed06a268dea0bc1e85/src/Npgsql/Npgsql.csproj#L34-L46

Since all this code _could_ be enabled by taking an additional dependency, I didn't know if it was worth the complexity of the conditional compilation when it could be the same for all TFMs.

Reply:
Yeah, what @bgrainger writes makes sense to me - better to simplify our code here and remove conditional compilation if we can, especially given that we already multi-target and bring in different packages based on TFM. For me, supporting old frameworks is mostly a bonus on top of that...